cmake/linker/ld/target_base.cmake

```diff
@@ -11,10 +11,15 @@ macro(toolchain_ld_base)
   # TOOLCHAIN_LD_FLAGS comes from compiler/gcc/target.cmake
   # LINKERFLAGPREFIX comes from linker/ld/target.cmake
   zephyr_ld_options(
-    -no-pie
     ${TOOLCHAIN_LD_FLAGS}
   )
 
+  if(CONFIG_BOARD_NATIVE_POSIX)
```

Comment:
first comment.
We should not filter this based on a board type.
It's to limiting regarding users ability to enable / disable this flag.

As minimum this should have a Kconfig and with a proper default value.
A given board can then define a different default.

The setting may be promptless to avoid users accidentially changing the setting.